public/pages/Indices/containers/Indices/Indices.tsx

```diff
@@ -211,25 +226,79 @@ export default class Indices extends Component<IndicesProps, IndicesState> {
     const filterIsApplied = !!search;
     const page = Math.floor(from / size);
 
-    const pagination: Pagination = {
+    const pagination: OuiDataGridPaginationProps = {
       pageIndex: page,
       pageSize: size,
       pageSizeOptions: DEFAULT_PAGE_SIZE_OPTIONS,
-      totalItemCount: totalIndices,
+      onChangeItemsPerPage: (newSize) => this.setState({ size: newSize }),
+      onChangePage: (newPage) => this.setState({ from: newPage * size }),
     };
 
-    const sorting: EuiTableSortingType<ManagedCatIndex> = {
-      sort: {
-        direction: sortDirection,
-        field: sortField,
-      },
+    const sorting: OuiDataGridSorting = {
+      onSort: this.onSort,
+      columns: this.state.sortField ? [{ id: this.state.sortField, direction: this.state.sortDirection }] : [],
     };
 
     const selection: EuiTableSelectionType<ManagedCatIndex> = {
       onSelectionChange: this.onSelectionChange,
     };
 
-    const { history } = this.props;
+    const renderCellValue = ({ rowIndex, columnId }) => {
+      const rowData = this.state.indices[rowIndex];
+
+      console.log(rowData);
+
+      switch (columnId) {
+        case "index":
+          return <IndexDetail {...this.props} index={rowData.index} />;
```

Comment:
Loading this IndexDetail component is the cause of endless call. @kohinoor98 can you reach out to @SuZhou-Joe the writter for some guidance here.

Reply:
Yup, will do. Thanks @bowenlan-amzn